Refactor/device setup (#607)
NordicSemiconductor/pc-nrfconnect-shared: Changelog.md

[Semantic Versioning](https://semver.org/spec/v2.0.0.html) but contrary to it
every new version is a new major version.

## 46 - 2023-05-19

### Added

-   Progress bar to device setup dialog

> -   Prepare device can be used from apps to be able program device at any point
>     such as Recover Hex for readback protected devices

Reviewer: ```suggestion
-   `prepareDevice` function is now and can be used to program devices on demand
```

> -   Custom devices setups can be now done by implementing IDeviceSetup and
>     passing it to `deviceSetup` array in `DeviceSetup`

Reviewer: ```suggestion
-   Custom device setups can be now done by implementing `IDeviceSetup` and
    passing it as the `deviceSetup` array in `DeviceSetup`
```
